RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -995,19 +995,67 @@ reco::Track TrackExtenderWithMTDT<TrackCollection>::buildTrack(const reco::Track
     double thiterror = -1.;
     bool validmtd = false;
 
-    //need to better handle the cases with >1 hit in MTD
+    size_t ifirst(0), ihitcount(0), ietlcount(0);
     for (auto const& hit : trajWithMtd.measurements()) {
-      bool ismtd = hit.recHit()->geographicalId().det() == DetId::Forward &&
-                   ForwardSubdetector(hit.recHit()->geographicalId().subdetId()) == FastTime;
-      if (ismtd) {
-        const MTDTrackingRecHit* mtdhit = static_cast<const MTDTrackingRecHit*>(hit.recHit()->hit());
-        thit = mtdhit->time();
-        thiterror = mtdhit->timeError();
-        validmtd = true;
-        break;
+      if (hit.recHit()->geographicalId().det() == DetId::Forward &&
+          ForwardSubdetector(hit.recHit()->geographicalId().subdetId()) == FastTime) {
+        ihitcount++;
+        if (ihitcount == 1 && ifirst == 0) {
+          ifirst = ihitcount;
+        }
```

Comment:
`ifirst` could be assigned only once, outside (after) the loop:
```suggestion
        if (ihitcount > 0) {
          ifirst = 1;
        }
```

Reply:
to be removed entirely (the logic as it is written now is useless)